### PR TITLE
ci: Validate Carthage

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Add samples to install the SDK via Carthage to validate it.

#skip-changelog

## :bulb: Motivation and Context

Fixes GH-1053

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Samples/Carthage-Validation/XCFramework/XCFramework/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        
+        SentrySDK.start { options in
+            options.dsn = "https://a92d50327ac74b8b9aa4ea80eccfb267@o447951.ingest.sentry.io/5428557"
+            options.debug = true
+            options.sessionTrackingIntervalMillis = 5_000
```
Comment (bruno-garcia) [inline Samples/Carthage-Validation/XCFramework/XCFramework/AppDelegate.swift:12]:
```suggestion
```

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
Build is green so I guess we're good to go?

A final 'test' could be to introduce a commit to this branch to break it in a way that would have gone undetected.

**Comment by bruno-garcia** [comment]
Oh:
image

Was this added then removed? Maybe we need to configure the repo to remove the 'requried' build in this case

**Comment by philipphofmann** [comment]
> Oh:
> image
> 
> Was this added then removed? Maybe we need to configure the repo to remove the 'requried' build in this case

Thanks, I updated the branch rule.

**Pull request merged**